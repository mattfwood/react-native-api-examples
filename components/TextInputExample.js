import React, { Component } from 'react';
import { View, TextInput, Text } from 'react-native';
import styled from 'styled-components';
import { Container } from './Layout';

const ExampleInput = styled.TextInput`
  background-color: #FFF;
  padding: 8px;
`;

const Heading = styled.Text`
  font-size: 24px;
  /* font-weight: 500; */
  margin-bottom: 10px;
`;

class TextInputExample extends Component {
  state = {
    value: '',
  }

  // onChangeText returns the value
  // event.target.value isn't needed
  onChange = (value) => {
    console.log('CHANGING INPUT');
    console.log(value);
    this.setState({ value });
  }

  render() {
    const { value } = this.state;
    return (
      <Container>
        <Heading>{`<TextInput />`}</Heading>
        <ExampleInput onChangeText={this.onChange} />
        <Text>Value: {value}</Text>
      </Container>
    );
  }
}

export default TextInputExample;