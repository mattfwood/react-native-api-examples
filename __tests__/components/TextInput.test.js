import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import TextInputExample from '../../components/TextInputExample';

it('renders correctly', () => {
  const tree = renderer.create(<TextInputExample />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('accepts user input', () => {

})