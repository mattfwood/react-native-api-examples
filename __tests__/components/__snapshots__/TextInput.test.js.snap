// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 15,
        "marginLeft": 15,
        "marginRight": 15,
        "marginTop": 15,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 24,
          "marginBottom": 10,
        },
      ]
    }
  >
    &lt;TextInput /&gt;
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
  <Text>
    Value: 
    
  </Text>
</View>
`;
